pkg/handlers/archive.go

```diff
@@ -153,23 +170,28 @@ func (d *Archive) ReadToMax(reader io.Reader) (data []byte, err error) {
 		}
 	}()
 	fileContent := bytes.Buffer{}
-	log.Tracef("Remaining buffer capacity: %d", d.maxSize-d.size)
-	for i := 0; i <= d.maxSize/512; i++ {
-		fileChunk := make([]byte, 512)
-		bRead, err := reader.Read(fileChunk)
-		if err != nil && !errors.Is(err, io.EOF) {
-			return []byte{}, err
-		}
-		d.size += bRead
-		if len(fileChunk) > 0 {
-			fileContent.Write(fileChunk[0:bRead])
-		}
-		if bRead < 512 {
-			break
-		}
-		if d.size >= d.maxSize && bRead == 512 {
-			log.Debug("Max archive size reached.")
-			break
+	log.Tracef("Remaining buffer capacity: %d", maxSize-d.size)
+	for i := 0; i <= maxSize/512; i++ {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+			fileChunk := make([]byte, 512)
+			bRead, err := reader.Read(fileChunk)
+			if err != nil && !errors.Is(err, io.EOF) {
+				return []byte{}, err
+			}
+			d.size += bRead
+			if len(fileChunk) > 0 {
+				fileContent.Write(fileChunk[0:bRead])
+			}
+			if bRead < 512 {
+				return fileContent.Bytes(), nil
+			}
+			if d.size >= maxSize && bRead == 512 {
+				log.Debug("Max archive size reached.")
+				return fileContent.Bytes(), nil
+			}
 		}
```

Comment:
**Nit (optional):** I think we can use `common.IsDone(ctx)` at the start of the loop and remove the `select`. That could help de-indent the body.